core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/tuple/Tuple.java

```diff
@@ -580,4 +584,56 @@ public Builder removeIfExists(String... attributes) {
         }
 
     }
+
+    public static class BuilderV2 {
+        private final Schema schema;
+        private final Map<String, Object> fieldNameMap;
+
+        public BuilderV2(Schema schema) {
+            this.schema = schema;
+            this.fieldNameMap = new HashMap<>();
+        }
+
+        public BuilderV2 fill(Tuple tuple) {
```

Comment:
can we add comments to say that schema is already set and the same names should be used etc.